flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphMetricsTest.java

```diff
@@ -140,6 +140,9 @@ public void testExecutionGraphRestartTimeMetric() throws JobException, IOExcepti
 
 				assertTrue(currentRestartingTime >= previousRestartingTime);
 				previousRestartingTime = currentRestartingTime;
+
+				// add some pause to let the currentRestartingTime increase
+				Thread.sleep(1L);
```

Comment:
move this to the start of the loop instead? (since you call `restartingTime.getValue()` right before the loop and at the start of the loop)

Reply:
I think we have to keep it here, because otherwise the loop might finish so fast that we don't see an increase in `previousRestartingTime` because this value is effectively `System.currentTimeMillis - timestamps[RESTARTING]`.

Reply:
Yes can do.